core/tests/puppeteer-acceptance-tests/user-utilities/logged-in-users-utils.ts

```diff
@@ -674,10 +674,11 @@ export class LoggedInUser extends BaseUser {
       throw new Error('The Watch A Video button does not exist!');
     }
     await Promise.all([
-      this.page.waitForNavigation(),
+      this.page.waitForNavigation({waitUntil: ['load', 'networkidle2']}),
       this.clickOn(watchAVideoButton),
     ]);
 
+    await this.waitForText('Create New Account');
```

Comment:
Why are we waiting for a certain text to be displayed instead of finding the relevant selector and using `waitForSelector` on it?

Reply:
@StephenYu2018 Actually, the page we’re being navigating to is a Facebook page. Here, we encounter selectors such as `class="x1ey2m1c x9f619 xds687c x10l6tqk x17qophe x13vifvy x1ypdohk"`. I guess, these are automatically generated ones and may not be reliable. 
Additionally, there are cases where it might be more reliable to wait for the text to appear.

Thank you.

Reply:
Why do you use the text "Create New Account"?

Reply:
Hm, for cases like this, I think what I would suggest is to just verify that the URL of the new page is correct and that the navigation stabilizes.

The reason I suggest that is because we don't want our tests to break if Facebook (say) changes how they render their page, which is something that's out of our control. Given that, do you think it would be enough to wait for navigation to stabilize?